### PR TITLE
Add react support

### PR DESCRIPTION
**Pull request by thornbill**
**Changes**
* Adds support for rendering pages as react components
* Converts the search page to use react

**Caveats**
* Our webcomponents are not working as they should currently within React components. This appears to be caused by an incompatibility between the polyfill we use due to our components using the now deprecated "v0" specification and React. I was able to work around this issue by using `dangerouslySetInnerHTML` to inject the webcomponents. In the future, we should move away from using those components since the spec is no longer supported.

**Why use React?**
* Allows us to use a modern frontend framework instead of building HTML strings and injecting them in the DOM.
* Well established library with good industry support and tooling.
* We already have a few projects in the organization using React (jellyfin-expo, jellyfin.org)
* Supports older browsers. (Unfortunately something like Vue 3 would not support older browsers used in TVs.)

**Issues**
N/A

### CODE REVIEW

**Comment by goldyydev** [comment]
Totally agree on moving to react, it would require a lot of code though.

**Comment by thornbill** [comment]
Right. The idea is rewriting existing pages could be done gradually over time and any new pages could use it immediately. No reason to rewrite everything at once. 😄

File: src/components/search/LiveTVSearchResults.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        EnableTotalRecordCount: false,
+        ImageTypeLimit: 1,
+        IncludePeople: false,
+        IncludeMedia: false,
+        IncludeGenres: false,
+        IncludeStudios: false,
+        IncludeArtists: false
+    });
+
+    // FIXME: This query does not support Live TV filters
```
Comment (thornbill) [inline src/components/search/LiveTVSearchResults.js:48]: Just as a note to reviewers, this is the current behavior, just added a note that it is not working as expected and will require either server or frontend changes to fix.

File: src/components/pages/SearchPage.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                    parentId={parentId}
+                />
+            }
+            <SearchResults
+                serverId={serverId || ApiClient.serverId()}
+                parentId={parentId}
+                collectionType={collectionType}
+                query={query}
+            />
+            <LiveTVSearchResults
```
[the remark this answers is not in the document]
Comment (thornbill) [reply src/components/pages/SearchPage.js:27]: That component only shows results in Live TV libraries. When searching in a Live TV library, you have some of the same rows (like Movies) that you would see normally, but they use a different query and display different information in the cards (like channel and air time). This seemed like the cleanest approach to manage that separation.

**Pull request merged**